### PR TITLE
KoreBuild Port and VSTS

### PR DESCRIPTION
**Pull request by anurse**
This is a bit of a beast, and it's in a bit of a work-in-progress state but I want to take the leap and start migration now. This converts the repo to use the KoreBuild build system used by other ASP.NET components.

A rundown of what's in here:
* Add VSTS build using the standard template from aspnet/BuildTools
* Add KoreBuild
  * Renamed "tests" to "test" to follow KoreBuild patterns
* Remove some extra projects/solutions:
  * Mono.Mobile.sln
  * Mono.sln
  * DailyTests (which were checking NuGet feeds, not something we really do anymore)
  * Extra Samples (we could bring them back elsewhere, or in a different folder, but they don't make sense in the main build):
    * Android Samples 
    * WP8
    * Windows 8 C# and JS
    * iOS
* Convert projects to SDK (where possible) and PackageReference:
  * WAP projects (LoadTestHarness, Samples, and Client.JS.Tests) remain as non-SDK projects but with PackageReference
* Remove old build scripts
* Merge clients into a single project targeting .NET Standard 2.0, .NET Framework 4.0 and .NET Framework 4.5
* Disabled IIS Express support in Functional Tests because they're broken
* Removed Functional Tests from the main project

TODO in this PR:
* [x] Resolve flaky tests
* [x] Clear up ruleset merge
* [x] Pack JS files into nupkg
* [x] Review produced nupkgs - Initial review shows they look the same, but I want to look deeper
  * [x] NuSpecs
  * [x] Assembly Paths
  * [x] Assemblies
  * [x] Resources

TODO after this PR (will file issues):
* [x] Re-enable Chutzpah-based JavaScript tests https://github.com/SignalR/SignalR/issues/4168
* [x] Re-enable Functional Tests (they run, they just take a long time) https://github.com/SignalR/SignalR/issues/4169
* [x] Release stuff (signing, loc, Intellisense Xml) https://github.com/SignalR/SignalR/issues/4171
* [x] Resolve Newtonsoft.Json dependency (NU1701 warning, etc.) https://github.com/SignalR/SignalR/issues/4170

### CODE REVIEW

**Comment by anurse** [comment]
@davidfowl I'm not really expecting you to be able to review the whole content of the PR :). Let's chat offline on this.

File: samples/Microsoft.AspNet.SignalR.LoadTestHarness/Microsoft.AspNet.SignalR.LoadTestHarness.csproj [hunk 1]
```diff
@@ -202,22 +199,14 @@
...
+    <ProjectReference Include="..\..\src\Microsoft.AspNet.SignalR.JS\Microsoft.AspNet.SignalR.JS.csproj" ReferenceOutputAssembly="false" />
   </ItemGroup>
-  <PropertyGroup>
-    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">10.0</VisualStudioVersion>
-    <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
-  </PropertyGroup>
   <Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />
-  <Import Project="$(VSToolsPath)\WebApplications\Microsoft.WebApplication.targets" Condition="'$(VSToolsPath)' != ''" />
-  <Import Project="$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v10.0\WebApplications\Microsoft.WebApplication.targets" Condition="false" />
+  <Import Project="$(WAPToolsPath)\WebApplications\Microsoft.WebApplication.targets" Condition="'$(VSToolsPath)' != ''" />
```
Comment (davidfowl) [inline samples/Microsoft.AspNet.SignalR.LoadTestHarness/Microsoft.AspNet.SignalR.LoadTestHarness.csproj]: @anurse I have an idea for this, ping me tomorrow.

**Comment by anurse** [comment]
I think this is ready to go. I'm looking in to the more complex release stuff (loc, signing) now but that'll be separate PRs as necessary

**Pull request merged**